File: code-challenges/arrayShift/array-shift.js
'use strict';

const insertShiftArray = (arr, val) => {
  let newArr = [];
  let addIdx = Math.ceil(arr.length / 2);
  arr.length === 0 ? newArr.push(val) : iterateArr(arr, val, addIdx, newArr);
  return newArr;
}

const iterateArr = (arr, val, idx, newArr) => {
  arr.forEach((elem, i) => {
    i !== idx ? newArr.push(elem) : (newArr.push(val), newArr.push(elem));
  });
}

module.exports = insertShiftArray;

File: code_challenges/array_shift/array_shift.py
import math

def iterate_arr(arr, val, add_indx, new_arr):
  for i, elem in enumerate(arr):
    (new_arr.append(val), new_arr.append(elem)) if i == add_indx else new_arr.append(elem)

def insert__shift_arr(arr,val):
  new_arr = []
  add_indx = math.ceil(len(arr) / 2)
  iterate_arr(arr, val, add_indx, new_arr) if len(arr) else new_arr.append(val)
  return new_arr

print(insert__shift_arr([1,2,4,5], 3))